src/test/java/teammates/sqlui/webapi/GetHasResponsesActionTest.java

```diff
@@ -0,0 +1,476 @@
+package teammates.sqlui.webapi;
+
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import java.time.Duration;
+import java.time.Instant;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.mockito.Mockito;
+import org.testng.annotations.AfterMethod;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+
+import teammates.common.util.Const;
+import teammates.storage.sqlentity.Course;
+import teammates.storage.sqlentity.FeedbackQuestion;
+import teammates.storage.sqlentity.FeedbackSession;
+import teammates.storage.sqlentity.Instructor;
+import teammates.storage.sqlentity.Student;
+import teammates.ui.output.HasResponsesData;
+import teammates.ui.webapi.EntityNotFoundException;
+import teammates.ui.webapi.GetHasResponsesAction;
+import teammates.ui.webapi.JsonResult;
+
+/**
+ * SUT: {@link GetHasResponsesAction}.
+ */
+public class GetHasResponsesActionTest extends BaseActionTest<GetHasResponsesAction> {
+
+    private Course typicalCourse;
+    private Instructor typicalInstructor;
+    private FeedbackSession typicalFeedbackSession;
+    private FeedbackQuestion typicalFeedbackQuestion;
+    private Student typicalStudent;
+
+    @Override
+    protected String getActionUri() {
+        return Const.ResourceURIs.HAS_RESPONSES;
+    }
+
+    @Override
+    protected String getRequestMethod() {
+        return GET;
+    }
+
+    @BeforeMethod
+    void setUpMethod() {
+        typicalCourse = getTypicalCourse();
+        typicalInstructor = getTypicalInstructor();
+        typicalFeedbackSession = getTypicalFeedbackSession(typicalCourse);
+        typicalFeedbackQuestion = getTypicalFeedbackQuestionForSession(typicalFeedbackSession);
+        typicalStudent = getTypicalStudent();
+    }
+
+    @AfterMethod
+    void tearDownMethod() {
+        Mockito.reset(mockLogic);
+    }
+
+    @Test
+    void testExecute_emptyParameters_throwsInvalidHttpParameterException() {
+        verifyHttpParameterFailure();
+    }
```

Comment:
maybe can combine with test on lines 427-431

Reply:
okay nvm, i see that both `checkSpecificAccessControl()` and `execute()` check for this